### ISSUE TITLE
Deleting user does not remove client application configuration

### ISSUE THREAD

**Issue by simonpoole** [openstreetmap/openstreetmap-website#2566]
Deleting a user (via the admin UI), does not remove any application OAuth configuration. I assume this is due to deleting not actually deleting but keeping the record and simply setting "deleted" status. 

I believe wrt the app configuration this is a bit dangerous as it allows a potentially malicious entry to remain even when one might believe that the issue has been resolved. So either the app entries should be really deleted or we need a method of deactivating the app entry and cascading it to the individual user configurations. IMHO the former is preferable except if there are record keeping requirements that would speak against doing that.

**Comment by tomhughes** [comment]
(a) There's no such thing as a perma blocked user.
(b) Is there any evidence that a token for a deleted user is actually able to do anything?

**Comment by simonpoole** [comment]
a) I was referring to a more abstract notion of that, I realize that it isn't actually a concept born out by the code.
b) Misunderstanding: this is about a user generating a client config for an app and that remaining active after they have been deleted, not about tokens for apps authorized by now deleted users.

In any case I think simply providing a manual "delete app configuration" is the better solution as it allows to decide if this is necessary on a case by case base.

**Comment by tomhughes** [comment]
I don't see how a manual system is going to work, or rather it is going to require an amount of development that we just don't have - the only real option here is either what we have now or deleting everything.

I mean there's nothing stopping manual deletion right now, except that it's not a question that will ever cross my mind when deleting a user.

**Comment by PasLoin** [comment]
> Is there any evidence that a token for a deleted user is actually able to do anything?

Have found a "use case" : 

https://osmcha.org/changesets/130421392/?aoi=75c0a968-0abe-47e0-81f6-f193c9e785f1

This user has deleting himself his account few day ago and recreate an another one and use it with ID editor. 
Today he use Josm with probably previous OAuth and he make this changeset. 

So impossible to view his new changesets using user history in OSM.
And also not removing access on  OAuth can be a security issue on another cases.

**Comment by tomhughes** [comment]
That's not the same thing at all - this ticket is about applications registered to the user not authorisations for applications.

**Comment by mmd-osm** [comment]
Good catch! I'm a bit surprised that this is even possible, both when creating a new changeset, as well as uploading some changes.

**Comment by mmd-osm** [comment]
So now that a user can delete their own account as self service, do we want to include "OAuth 2 applications" in the list of  automatically removed items?
